### PR TITLE
Introduce developer guidelines

### PR DESCRIPTION
**Pull request by henn**

### CODE REVIEW

**Comment by genagain** [comment]
The guidelines look clear to me!

File: docs/developer-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+Once the checklist above has been met, issue a [pull request][pr] from your
+personal fork to the [main HaaS repo][repo]. 
+
+Labels are the way we communicate github pull request and issue status. They should be assigned in line with the *Suggestions* section
+of [this wiki
+page](https://github.com/CCI-MOC/haas/wiki/Issue-Tracker-Proposal#suggestions).
+
+## Code review
+
```
Comment (gsilvis) [inline docs/developer-guidelines.md]: Should there be something in this section?

**Comment by henn** [comment]
Could folks take another look at this? It should incorporate everyone's feedback now.

**Comment by SahilTikale** [comment]
@henn 
2nd paragraph under section `overview`, I think that content should be a separate section about "how to submit Pull Request to HaaS "

**Comment by SahilTikale** [comment]
In Section `Prior to a pull request`
I suggest providing an explicit to-do list. 
eg. Testing:
 `Run py.test tests/unit/ and make sure all the tests pass.`
`Add unit tests in the corresponding file and create one is none is present`

Also, mention about updating the rest_api.md whenever there is a PR involving API change.

**Comment by SahilTikale** [comment]
for section `Get Agreement` 
is there a way to hyperlink 'core developers' to somehow show who they are and their email addresses.
Should we have a single email address for the core-developer-group ?

**Comment by SahilTikale** [comment]
`section Documentation`
Should accompany some examples. 
eg. If a new api is introduced with corresponding CLI, update `rest_api.md` with the api related info and update `using.rst` with an example on how to use it via CLI and directly using API.

**Comment by SahilTikale** [comment]
`section Testing`

What does the stress.py do ? 
When do developers need to use it ?

**Comment by SahilTikale** [comment]
`Section Core developers`

What does PTL stand for ?

File: docs/developer-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
+# Overview
+
+This document represents the developer policies and procedures for maintaining
+Hardware as a Service (HaaS), including the accepted [coding style](#Coding-style).
+
+Welcome! Overall, we follow the [github fork & pull
+model](http://scottchacon.com/2011/08/31/github-flow.html), where users fork
+the [main HaaS repository][repo], push changes to their personal fork and then
```
Comment (xuhang57) [inline docs/developer-guidelines.md]: Should we mention that the user should create a new branch other than using the master branch in their personal fork each time when the user trying to make changes?

File: docs/developer-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+some cases, such as missing tests, it may help a new contributor for an
+experienced reviewer to work closely or even write the initial tests
+themselves.
+
+As a practical matter, if feedback is lengthy and would take a longer time to
+write up than simply to do, it can be considered a courtesy to check out the
+submitter's branch, make the changes yourself, then submit a Pull Request
+directly to the submitter's branch. This is especially helpful for
+documentation PRs. When doing so, it can help to notify the submitter directly
+as they may otherwise your pending PR.
```
Comment (pjd-nu) [inline docs/developer-guidelines.md]: "they may otherwise your pending PR" - missing word here.

**Comment by pjd-nu** [comment]
+1

**Comment by SahilTikale** [comment]
@henn how are we coming on this ?

**Comment by SahilTikale** [comment]
@henn any updates ?

**Comment by henn** [comment]
I believe I've incorporated all of the feedback. Review away @SahilTikale @pjd-nu @genagain @gsilvis!

**Comment by pjd-nu** [comment]
Looks good to me.

File: docs/developer-guidelines.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+documentation (linked from the [README](../README.rst)).
+
+If you're looking to code, there are a few ways to help:
+* Improving the HaaS documentation
+* Better testing
+* Reviewing [Pull Requests](https://github.com/CCI-MOC/haas/pulls)
+* Taking a github issue marked with the [easy label](https://github.com/CCI-MOC/haas/labels/easy). (Talk to one of the core team members to have it assigned to you officially!)
+* Taking on a github issue that is part of one of the next [milestones in github](https://github.com/CCI-MOC/haas/milestones?state=open)
+
+## Commuinicating
```
Comment (SahilTikale) [inline docs/developer-guidelines.md]: typo

File: docs/developer-guidelines.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+The exception to this is documentation, where we are a little more lenient in
+the interest of lowering the barrier to having better docs. For docs changes,
+typically one +1 is sufficient.
+
+Whomever provides the enabling +1 is responsible for clicking the merge button
+on github. If you do not have commit access, then please add your +1 to the PR
+and ask one of the Core Developers to complete the merge.
+
+## Labels
```
Comment (SahilTikale) [inline docs/developer-guidelines.md]: Also add a  line about "assignee".
It is good to assign an issue or PR to yourself, so that people know who is working on it.

Comment (henn) [inline docs/developer-guidelines.md]: I like the idea of having an assignee, however github only allows one and we typically require two reviewers.

Additionally, it's not necessarily a bad things if more than 1 or 2 reviewers review a request.

**Comment by SahilTikale** [comment]
+1 and merging.
